### PR TITLE
Fix user lookup by email for inbound mail

### PR DESCRIPTION
**Pull request by lukasgraf**
To avoid fuzzy matches when looking up the user by email address for inbound mails, the `pas_search` call [was extended with `exact_match=True`](https://github.com/4teamwork/opengever.core/commit/28d28fda2c1753d81662a4639eff123c86b0938e) to accomplish this.

However, it turns out that the `ActiveDirectoryMultiPlugin` [refuses to do searches](https://github.com/4teamwork/Products.LDAPMultiPlugins/blob/3c8a9aeaa125e23de919312891942bb07ead5ffb/Products/LDAPMultiPlugins/ActiveDirectoryMultiPlugin.py#L341-L348) with `exact_match=True`, unless the query criteria contain the user's id or login. For the `LDAPMultiPlugin` this [works fine](https://github.com/4teamwork/Products.LDAPMultiPlugins/blob/3c8a9aeaa125e23de919312891942bb07ead5ffb/Products/LDAPMultiPlugins/LDAPMultiPlugin.py#L188-L205), but the AD plugin raises a ValueError that later gets supressed, and so no matches are returned.

We therefore change the strategy to let `pas_search` return fuzzy matches again, and filter them post-search to only pick case-insensitive, but otherwise exact matches.

For [CA-4607](https://4teamwork.atlassian.net/browse/CA-4607)

## Checklist

- [x] Changelog entry
- [x] Link to issue (Jira or GitHub) and backlink in issue (Jira)

⚠️ Needs backport to `2022.15.x` ⚠️

### CODE REVIEW

File: opengever/mail/browser/inbound.py [hunk 1]
```diff
@@ -72,3 +84,25 @@ def get_user(self):
         log.warn('Sender address: %r' % sender_email)
         log.warn('No users found in %r for that address.' % acl_users)
         raise UnknownSender(self.msg())
+
+    def _get_email(self, acl_users, user_record):
+        """Get email address from a pas_search result record, while respecting
+        a potential schema mapping.
+
+        Unfortunately, pas_search result records contain unmapped attribute
+        names (raw, as the appear in LDAP/AD). It accepts mapped names
```
Comment (njohner) [inline opengever/mail/browser/inbound.py:93]: as the appear -> as they appear

File: opengever/mail/browser/inbound.py [hunk 2]
```diff
@@ -72,3 +84,25 @@ def get_user(self):
...
+        its schema, and look up the email address using the proper name.
+        """
+        email_attr_name = 'email'
+
+        plugin = acl_users[user_record['pluginid']]
+        if ILDAPMultiPlugin.providedBy(plugin):
+            schema = plugin._getLDAPUserFolder().getSchemaDict()
+            for entry in schema:
+                if entry['public_name'] == 'email':
+                    email_attr_name = entry['ldap_name']
```
Comment (njohner) [inline opengever/mail/browser/inbound.py:106]: I think you can add a `break` here.

**Review by njohner — APPROVED** [review]
LGTM, just wondering how painful it would be to add a regression test?

**Comment by lukasgraf** [comment]
@njohner I've had to change the strategy a bit again, would you mind doing another review?
_(Need to fetch the actual user for each potential match, because only `user.getProperty('email')` works reliable. For the AD plugin, the email isn't returned in the result records)._

I've tested the code above as-is on DIGS TEST in SG, and it does work there.

**Review by njohner — APPROVED** [review]
LGTM.
I think that this code will not be necessary anymore once we implement our OGDS pas plugin no? Maybe you can add a comment in the corresponding Jira issue (well I'm not sure where that would be, I don't think any story covers actually using the new plugin, but somewhere here https://4teamwork.atlassian.net/browse/ROAD-2458)

**Pull request merged**